### PR TITLE
OPEN-444 automated export of all comments

### PR DESCRIPTION
**Pull request by HoussamBedja**

### CODE REVIEW

File: modules/custom/external_comment/external_comment.module [hunk 1]
```diff
@@ -25,3 +25,78 @@ function external_comment_query_comment_filter_alter(AlterableInterface $query)
     }
   }
 }
+
+
+
```
Comment (RabiaSajjad) [inline modules/custom/external_comment/external_comment.module]: please add annotations

File: modules/custom/external_comment/external_comment.module [hunk 2]
```diff
@@ -25,3 +25,78 @@ function external_comment_query_comment_filter_alter(AlterableInterface $query)
...
+function external_comment_cron() {
+  // fetch comments
+  $comment_ids = \Drupal::entityQuery('comment')
```
Comment (RabiaSajjad) [inline modules/custom/external_comment/external_comment.module]: you may want to start with nodes?

File: modules/custom/external_comment/external_comment.module [hunk 3]
```diff
@@ -25,3 +25,78 @@ function external_comment_query_comment_filter_alter(AlterableInterface $query)
...
+    $comments = \Drupal::entityTypeManager()
+      ->getStorage('comment')
+      ->loadMultiple($comment_ids);
+
+    foreach($comments as $comment) {
+      // Loop over and get fields for published comments
+      if ($comment->getStatus() == 1) {
+        $comments_data[] = [
+          'comment_id' => $comment->id(),
+          'uuid' => $comment->uuid(),
```
Comment (RabiaSajjad) [inline modules/custom/external_comment/external_comment.module]: this is comment entity's uuid, what you are looking for is content-type 'external'->field_uuid

File: modules/custom/external_comment/external_comment.module [hunk 4]
```diff
@@ -25,3 +25,78 @@ function external_comment_query_comment_filter_alter(AlterableInterface $query)
...
+          'comment_posted_by' => $comment->getAuthorName(),
+          'date_posted' => \Drupal::service('date.formatter')->format($comment->getCreatedTime(), 'html_date')
+        ];
+      }
+    }
+  }
+
+  $header = [
+    'Comment id',
+    'Page uuid',
```
Comment (RabiaSajjad) [inline modules/custom/external_comment/external_comment.module]: may want to add actual page where this comment was posted

Comment (RabiaSajjad) [inline modules/custom/external_comment/external_comment.module]: for content type 'external' link to page is saved in field_url

File: modules/custom/external_comment/external_comment.module [hunk 5]
```diff
@@ -25,3 +25,78 @@ function external_comment_query_comment_filter_alter(AlterableInterface $query)
...
+
+  // export as csv
+  write_to_csv('comments.csv', $comments_data, $header);
+}
+
+
+/**
+ * Generate output file for given data and headers
+ */
+function write_to_csv($filename, $data_to_write, $csv_header) {
```
Comment (RabiaSajjad) [inline modules/custom/external_comment/external_comment.module]: if you have only one function writing to a csv, you may include it in the function itself?

File: modules/custom/external_comment/external_comment.module (hunk 2 above)
Comment (HoussamBedja) [reply modules/custom/external_comment/external_comment.module]: I don't fully understand this, but let me know if it's still applicable after the new changes.

**Comment by RabiaSajjad** [comment]
Looking good @HoussamBedja, let's merge to staging and test with @PatLittle

**Comment by RabiaSajjad** [comment]
Great! thanks for the hard work

**Pull request merged**